server/src/tables.rs

```diff
@@ -403,140 +547,241 @@ impl ExecutionPlan for GridExec {
+        cx: &mut StdTaskContext<'_>,
+    ) -> Poll<Option<Self::Item>> {
         let poll = self.input.poll_next_unpin(cx).map(|x| match x {
-            Some(Ok(batch)) => Some(self.grid(&batch)),
+            Some(Ok(batch)) => Some(Ok(self.grid(&batch))),
             other => other,
         });
         self.baseline_metrics.record_poll(poll)
     }
 }
 
 impl RecordBatchStream for GridStream {
+    /// Return the schema of the model table after projection.
     fn schema(&self) -> SchemaRef {
         self.schema_after_projection.clone()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use datafusion::logical_plan::lit;
+    use datafusion::prelude::Expr;
+
+    // Tests for rewrite_and_combine_filters().
+    #[test]
+    fn test_rewrite_empty_vec() {
+        assert!(rewrite_and_combine_filters(&vec!()).is_none());
+    }
+
+    #[test]
+    fn test_rewrite_greater_than_timestamp() {
+        let filters = new_timestamp_filters(Operator::Gt);
+        let predicate = rewrite_and_combine_filters(&filters).unwrap();
+        assert_timestamp_expr(predicate, "end_time", Operator::Gt);
+    }
+
+    #[test]
+    fn test_rewrite_greater_than_or_equal_timestamp() {
+        let filters = new_timestamp_filters(Operator::GtEq);
+        let predicate = rewrite_and_combine_filters(&filters).unwrap();
+        assert_timestamp_expr(predicate, "end_time", Operator::GtEq);
+    }
+
+    #[test]
+    fn test_rewrite_less_than_timestamp() {
+        let filters = new_timestamp_filters(Operator::Lt);
+        let predicate = rewrite_and_combine_filters(&filters).unwrap();
+        assert_timestamp_expr(predicate, "start_time", Operator::Lt);
+    }
+
+    #[test]
+    fn test_rewrite_less_than_or_equal_timestamp() {
+        let filters = new_timestamp_filters(Operator::LtEq);
+        let predicate = rewrite_and_combine_filters(&filters).unwrap();
+        assert_timestamp_expr(predicate, "start_time", Operator::LtEq);
+    }
+
+    #[test]
+    fn test_rewrite_equal_timestamp() {
+        let filters = new_timestamp_filters(Operator::Eq);
+        let predicate = rewrite_and_combine_filters(&filters).unwrap();
+
+        if let Expr::BinaryExpr { left, op, right } = predicate {
+            assert_timestamp_expr(*left, "start_time", Operator::LtEq);
+            assert_eq!(op, Operator::And);
+            assert_timestamp_expr(*right, "end_time", Operator::GtEq);
+        } else {
+            panic!("Expr is not a BinaryExpr");
+        }
+    }
+
+    fn new_timestamp_filters(operator: Operator) -> Vec<Expr> {
+        vec![new_binary_expr(col("timestamp"), operator, lit(37))]
+    }
+
+    fn assert_timestamp_expr(expr: Expr, column: &str, operator: Operator) {
+        if let Expr::BinaryExpr { left, op, right } = expr {
+            assert_eq!(*left, col(column));
+            assert_eq!(op, operator);
+            assert_eq!(*right, lit(37));
+        } else {
+            panic!("Expr is not a BinaryExpr");
```

Comment:
Very minor but this panic message is missing punctuation. Should also be fixed on line 741.

Reply:
Fixed.